torch/csrc/jit/tensorexpr/tensorexpr_init.cpp

```diff
@@ -430,6 +431,54 @@ void initTensorExprBindings(PyObject* module) {
       [](Stmt* stmt) { return IRSimplifier::simplify(stmt); },
       py::return_value_policy::reference);
 
+  using TSGraph = std::shared_ptr<Graph>;
+  py::class_<TensorExprKernel>(te, "TensorExprKernel")
+      .def(py::init<const TSGraph>())
+      .def_static(
+          "run",
+          [](const TSGraph graph, const py::tuple& inputs) {
+            Stack stack;
+            stack.reserve(inputs.size()); // captures?
+            for (auto& obj : inputs) {
+              stack.push_back(toTypeInferredIValue(obj));
+            }
+            auto g_inputs = graph->inputs();
+            for (size_t i = 0; i < inputs.size(); ++i) {
+              if (stack[i].isTensor()) {
+                g_inputs[i]->setType(stack[i].type());
+              }
+            }
+            TensorExprKernel(graph).run(stack);
```

Comment:
When preparing a stack in `run`, it needs the inputs of the graph but `TensorExprKernel` does not have any APIs for that yet. If we separate the construction of `TensorExprKernel` from `run`, we need to add new public APIs in `TensorExprKernel` c++ code to expose the graph or the inputs. 
So the options are, 
1) add `TensorExprKernel::get_inputs()` or  `TensorExprKernel::get_graph()`
2) let the user pass in the `graph` parameter again when calling `run()` which seems redundant to the user because the user has constructed a kernel with the `graph` parameter
3) like what I did - put `run()` and `fallback()` as static methods which would construct the kernel itself (no redundancy) and run/fallback it

Reply:
Please add `TensorExprKernel::graph()` API to retrieve the graph for the kernel. Whatever we do we need to keep in mind the main purpose of the API we're introducing. In this case, `run` would most probably be used for benchmarking and having to compile the kernel each time we want to run it is a very bad idea for performance.